test/common/secret/secret_manager_impl_test.cc

```diff
@@ -50,6 +50,82 @@ name: "abc.com"
             secret_manager->findStaticTlsCertificateProvider("abc.com")->secret()->privateKey());
 }
 
+TEST_F(SecretManagerImplTest, DuplicateStaticTlsCertificateSecret) {
+  envoy::api::v2::auth::Secret secret_config;
+
+  std::string yaml =
+      R"EOF(
+    name: "abc.com"
+    tls_certificate:
+      certificate_chain:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned_cert.pem"
+      private_key:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned_key.pem"
+    )EOF";
+
+  MessageUtil::loadFromYaml(TestEnvironment::substitute(yaml), secret_config);
+
+  std::unique_ptr<SecretManager> secret_manager(new SecretManagerImpl());
+
+  secret_manager->addStaticSecret(secret_config);
+
+  ASSERT_NE(secret_manager->findStaticTlsCertificateProvider("abc.com"), nullptr);
+
+  EXPECT_THROW_WITH_MESSAGE(secret_manager->addStaticSecret(secret_config), EnvoyException,
+                            "Duplicate static TlsCertificate secret name abc.com");
+}
+
+TEST_F(SecretManagerImplTest, CertificateValidationContextSecretLoadSuccess) {
+  envoy::api::v2::auth::Secret secret_config;
+
+  std::string yaml =
+      R"EOF(
+      name: "abc.com"
+      validation_context:
+        trusted_ca: { filename: "{{ test_rundir }}/test/common/ssl/test_data/ca_cert.pem" }
+        allow_expired_certificate: true
+      )EOF";
+
+  MessageUtil::loadFromYaml(TestEnvironment::substitute(yaml), secret_config);
+
+  std::unique_ptr<SecretManager> secret_manager(new SecretManagerImpl());
+
+  secret_manager->addStaticSecret(secret_config);
+
+  ASSERT_EQ(secret_manager->findStaticCertificateValidationContextProvider("undefined"), nullptr);
+
+  ASSERT_NE(secret_manager->findStaticCertificateValidationContextProvider("abc.com"), nullptr);
+
+  const std::string cert_pem = "{{ test_rundir }}/test/common/ssl/test_data/ca_cert.pem";
+  EXPECT_EQ(TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(cert_pem)),
+            secret_manager->findStaticCertificateValidationContextProvider("abc.com")
+                ->secret()
+                ->caCert());
+}
+
+TEST_F(SecretManagerImplTest, DuplicateStaticCertificateValidationContextSecret) {
```

Comment:
Can you add a one line description of what each test does above it.

Reply:
Comments are added above each test. Thanks.